src/libs/ajax/AzureStorage.ts

```diff
@@ -111,6 +111,54 @@ export const AzureStorage = (signal?: AbortSignal) => ({
     }), notebooks)
   },
 
+  blobMetadata: (azureStorageUrl: string) => {
+    const getObjectMetadata = async () => {
+      const workspaceId = azureStorageUrl.split('/')[3].replace('sc-', '')
+      const fileName = _.last(azureStorageUrl?.split('/'))?.split('.')?.join('.')
+
+      try {
+        // assumption is made that container name guid in uri always matches the workspace Id guid it is present in
+        const { sas: { token } } = await AzureStorage(signal).details(workspaceId)
+
+        // instead of taking the url returned by azure storage, take it from the incoming url since there may be a folder path
+        const urlwithFolder = azureStorageUrl.substring(0, azureStorageUrl.lastIndexOf('/'))
+        const azureSasStorageUrl = _.flow(
```

Comment:
I'm not entirely sure what's going on here. It might be clearer to create a [URL](https://developer.mozilla.org/en-US/docs/Web/API/URL) here and replace parts of it (path and/or query string) instead of using string methods to parse the URL.

Reply:
using URL makes a whole lot more sense, adjusted!